sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezZPageHandler.java

```diff
@@ -0,0 +1,603 @@
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.Status.CanonicalCode;
+import java.io.BufferedWriter;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.Serializable;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Formatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+final class TracezZPageHandler extends ZPageHandler {
+  private enum SampleType {
+    RUNNING(0),
+    LATENCY(1),
+    ERROR(2),
+    UNKNOWN(-1);
+
+    private final int value;
+
+    SampleType(int value) {
+      this.value = value;
+    }
+
+    static SampleType fromString(String str) {
+      int value = Integer.parseInt(str);
+      switch (value) {
+        case 0:
+          return RUNNING;
+        case 1:
+          return LATENCY;
+        case 2:
+          return ERROR;
+        default:
+          return UNKNOWN;
+      }
+    }
+
+    int getValue() {
+      return value;
+    }
+  }
+
+  private static final String TRACEZ_URL = "/tracez";
+  // Background color used for zebra striping rows of summary table
+  private static final String ZEBRA_STRIPE_COLOR = "#e6e6e6";
+  // Color for sampled traceIds
+  private static final String SAMPLED_TRACE_ID_COLOR = "#c1272d";
+  // Color for not sampled traceIds
+  private static final String NOT_SAMPLED_TRACE_ID_COLOR = "black";
+  // Query string parameter name for span name
+  private static final String PARAM_SPAN_NAME = "zspanname";
+  // Query string parameter name for type to display
+  // * 0 = running, 1 = latency, 2 = error
+  private static final String PARAM_SAMPLE_TYPE = "ztype";
+  // Query string parameter name for sub-type:
+  // * for latency based sampled spans [0, 8] corresponds to each latency boundaries
+  //   where 0 corresponds to the first boundary
+  // * for error based sampled spans [0, 15], 0 means all, otherwise the error code
+  private static final String PARAM_SAMPLE_SUB_TYPE = "zsubtype";
+  @javax.annotation.Nullable private final TracezDataAggregator dataAggregator;
+  // Map from LatencyBoundaries to human readable string on the UI
+  private static final ImmutableMap<LatencyBoundaries, String> LATENCY_BOUNDARIES_STRING_MAP =
+      buildLatencyBoundariesStringMap();
+
+  private TracezZPageHandler(TracezDataAggregator dataAggregator) {
+    this.dataAggregator = dataAggregator;
+  }
+
+  /**
+   * Constructs a new {@code TracezZPageHandler}.
+   *
+   * @return a new {@code TracezZPageHandler}.
+   */
+  static TracezZPageHandler create(@javax.annotation.Nullable TracezDataAggregator dataAggregator) {
```

Comment:
The class is package private, I think we can make the constructor package private and remove this factory